pkg/telemetrymetadata/metadata.go

```diff
@@ -316,6 +316,55 @@ func (t *telemetryMetaStore) getTracesKeys(ctx context.Context, fieldKeySelector
 			})
 		}
 	}
+	presentByName := map[string]*telemetrytypes.TelemetryFieldKey{}
+	for _, k := range keys {
+		if _, ok := presentByName[k.Name]; !ok {
+			presentByName[k.Name] = k
+		}
+	}
+
+	findMaterialized := func(name string) *telemetrytypes.TelemetryFieldKey {
+		for _, mk := range matKeys {
+			if mk.Name == name {
+				return mk
+			}
+		}
+		return nil
+	}
+
+	appendAliasIfMissing := func(alias string, src *telemetrytypes.TelemetryFieldKey) {
+		if mk := findMaterialized(alias); mk != nil {
+			keyID := mk.Name + ";" + mk.FieldContext.StringValue() + ";" + mk.FieldDataType.StringValue()
+			if _, exists := mapOfKeys[keyID]; !exists {
+				aliasKey := *mk
+				aliasKey.Signal = telemetrytypes.SignalTraces
+				keys = append(keys, &aliasKey)
+				mapOfKeys[keyID] = &aliasKey
+				presentByName[alias] = &aliasKey
+			}
+			return
+		}
+
+		aliasKey := &telemetrytypes.TelemetryFieldKey{
+			Name:          alias,
+			Signal:        telemetrytypes.SignalTraces,
+			FieldContext:  src.FieldContext,
+			FieldDataType: src.FieldDataType,
+		}
+		keyID := aliasKey.Name + ";" + aliasKey.FieldContext.StringValue() + ";" + aliasKey.FieldDataType.StringValue()
+		if _, exists := mapOfKeys[keyID]; !exists {
+			keys = append(keys, aliasKey)
+			mapOfKeys[keyID] = aliasKey
+			presentByName[alias] = aliasKey
+		}
+	}
+
+	// Add backward compatible aliases for traces
+	for srcKey, aliasKey := range TracesBackwardCompatKeys {
```

Comment:
This entire part is not required right ? since we are already adding it using applyBackwardCompatibleKeys